File: rachinha_back/app/interfaces/schemas/feedback.py
from pydantic import BaseModel, Field
from typing import Literal

class Feedback(BaseModel):
    title: str = Field(..., description="Título do feedback")
    description: str = Field(..., description="Descrição do feedback")
    type: Literal["suggestion", "bug", "other"] = Field(..., description="Tipo de feedback")

File: rachinha_front/services/feedback.ts
import api from "./api";
import { handleApiError } from "@/lib/utils";

export interface Feedback {
  title: string;
  description: string;
  type: "bug" | "suggestion" | "other";
}

export async function submitFeedback(feedback: Feedback): Promise<void> {
  try {
    await api.post("/feedback/", feedback);
  } catch (error) {
    throw handleApiError(error, "Erro ao enviar feedback");
  }
}
